Infrastructure-Scripts/Set-AppRoleAssignments/Set-AppRoleAssignments.ps1

```diff
@@ -92,18 +93,18 @@ try {
                 $ServicePrincipal = Get-ServicePrincipal -DisplayName $AppRegistrationName
                 $MatchedAppRole = $ServicePrincipal.appRoles | Where-Object { $_.value -eq $AppRole.appRoleName }
             }
-            foreach ($MIResource in $AppRole.resourceNameSuffix) {
-                $ManagedIdentityResourceName = "$ResourceNamePrefix$MIResource"
-                $ManagedIdentity = Get-ServicePrincipal -DisplayName $ManagedIdentityResourceName
+
+            if ($ResourceNameSuffix -in $AppRole.resourceNameSuffix) {
+                $ManagedIdentity = Get-ServicePrincipal -DisplayName $ResourceName
                 if ($ManagedIdentity.Count -eq 1) {
-                    Write-Output "    -> Processing Managed Identity of $ManagedIdentityResourceName"
+                    Write-Output "    -> Processing Managed Identity of $ResourceName"
                 }
                 elseif ($ManagedIdentity.Count -gt 1) {
-                    Write-Error "    -> Found duplicate app registrations with the same display name of $ManagedIdentityResourceName. Investigate"
+                    Write-Error "    -> Found duplicate app registrations with the same display name of $ResourceName. Investigate"
                     continue
                 }
                 else {
-                    Write-Output "    -> Managed Identity of $ManagedIdentityResourceName not found"
+                    Write-Output "    -> Managed Identity of $ResourceName not found"
                     continue
                 }
 
```

Comment:
Does the `if ($ServicePrincipal.ObjectId)` condition still need to exist?  The change you've made means it no longer has a corresponding else block.  It looks to me like the try statement and this condition might be trying to achieve the same thing

Reply:
You're right that it no longer needs to exist. The script would fail much earlier as well if that object didnt exist anyway. It was a condition that was made redundant after the new changes 👍